umongo/frameworks/pymongo.py

```diff
@@ -322,11 +322,14 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._document = None
 
-    def fetch(self, no_data=False, force_reload=False):
+    def fetch(self, no_data=False, force_reload=False, projection=None):
         if not self._document or force_reload:
             if self.pk is None:
                 raise NoneReferenceError('Cannot retrieve a None Reference')
-            self._document = self.document_cls.find_one(self.pk)
+            if projection is None:
+                self._document = self.document_cls.find_one(self.pk)
+            else:
+                self._document = self.document_cls.find_one(self.pk, projection)
```

Comment:
Why not a single case

    self._document = self.document_cls.find_one(self.pk, projection=projection)

Doesn't this work for all cases?